spec/controllers/campaigns_controller_spec.rb

```diff
@@ -64,16 +64,27 @@
     let(:user) { create(:user) }
     let(:admin) { create(:admin) }
     let(:campaign) { create(:campaign) }
+    let(:campaign_user) do
+      create(:campaigns_user, user_id: user.id, campaign_id: campaign.id,
+                               role: CampaignsUsers::Roles::ORGANIZER_ROLE)
+    end
     let(:description) { 'My new campaign is the best campaign ever!' }
     let(:campaign_params) { { slug: campaign.slug, description: description } }
 
-    it 'returns a 401 if the user is not an admin and feature flag is off' do
+    it 'returns a 401 if the user is not an admin and not an organizer of the campaign' do
       allow(controller).to receive(:current_user).and_return(user)
       allow(Features).to receive(:open_course_creation?).and_return(false)
       post :update, params: { campaign: campaign_params, slug: campaign.slug }
       expect(response.status).to eq(401)
     end
 
+    it 'updates the campaign if the user is an organizer of the campaign' do
+      allow(controller).to receive(:current_user).and_return(campaign_user)
```

Comment:
You need to create the CampaignsUsers record first, but you still want :current_user to return :user.